File: backend/src/Alife.Domain/Entities/Announcement.cs
using Alife.Domain.Enums;

namespace Alife.Domain.Entities;

public class Announcement
{
    public Guid Id { get; set; }
    public Guid GroupId { get; set; }
    public string TitleJson { get; set; } = "{}";
    public string SummaryJson { get; set; } = "{}";
    public string? ContentJson { get; set; }
    public AnnouncementAudience Audience { get; set; }
    public AnnouncementPriority Priority { get; set; }
    public AnnouncementStatus Status { get; set; }
    public DateTime PublishUtc { get; set; }
    public DateTime? ExpireUtc { get; set; }
    public bool IsPinned { get; set; }
    public Guid CreatedByMemberId { get; set; }
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }

    public Group Group { get; set; } = null!;
    public Member CreatedByMember { get; set; } = null!;
    public ICollection<NotificationMessage> Notifications { get; set; } = [];
}

File: cloudflare/alife-app/src/types/announcement.ts
import type { LocalizedText } from './models'

export type AnnouncementAudience = 'public' | 'churchMembers' | 'specificGroup'
export type AnnouncementPriority = 'normal' | 'important' | 'urgent'
export type AnnouncementStatus = 'draft' | 'published' | 'archived'

export type AnnouncementDto = {
  id: string
  groupId: string
  title: LocalizedText
  summary: LocalizedText
  content?: LocalizedText | null
  audience: AnnouncementAudience
  priority: AnnouncementPriority
  status: AnnouncementStatus
  publishUtc: string
  expireUtc?: string | null
  isPinned: boolean
  createdByMemberId: string
  createdUtc: string
  updatedUtc: string
}

export type SaveAnnouncementPayload = Pick<AnnouncementDto,
  'groupId' | 'title' | 'summary' | 'content' | 'audience' | 'priority' | 'status' | 'publishUtc' | 'expireUtc' | 'isPinned'> & {
  createNotifications: boolean
}
